### PR TITLE
Add priority

### PR DESCRIPTION
**Pull request by ismailgulek**
Added `priority` into `Rule`, which was implemented in the [json-rules-engine](https://github.com/CacheControl/json-rules-engine/tree/master).
Documentation is here: https://github.com/CacheControl/json-rules-engine/blob/master/docs/rules.md#methods

### CODE REVIEW

**Comment by ismailgulek** [comment]
@santalvarez it'd be great if you can review and merge this.

**Comment by santalvarez** [comment]
Hey! Thanks a lot for contributing. I will review it today.

File: Tests/SwiftRuleEngineTests/RuleEngineTests/RuleEngineDictionaryTests.swift [hunk 1]
```diff
@@ -376,4 +376,50 @@ class RuleEngineTests: XCTestCase {
 
         XCTAssertTrue(result.conditions.match)
     }
+
+    func testRulePriority() throws {
+        let rule1: [String: Any] = [
+            "name": "test-rule",
```
Comment (santalvarez) [inline Tests/SwiftRuleEngineTests/RuleEngineTests/RuleEngineDictionaryTests.swift:382]: I don't think we are testing the priority correctly in this test. To correctly test it the two rules should be identical and the priority of the second one should be higher. Then we test that the second one matched and not the first one so we can make sure that the rules were correctly sorted.
As it is right now the rule 1 will never match even with no priority. Lets make that change and I'll merge

**Review by santalvarez — CHANGES_REQUESTED** [review]

File: Tests/SwiftRuleEngineTests/RuleEngineTests/RuleEngineDictionaryTests.swift (hunk 1 above)
Comment (ismailgulek) [reply Tests/SwiftRuleEngineTests/RuleEngineTests/RuleEngineDictionaryTests.swift:382]: That's what I exactly thought but set the priority wrong 🤦‍♂️, updated now

Comment (santalvarez) [reply Tests/SwiftRuleEngineTests/RuleEngineTests/RuleEngineDictionaryTests.swift:382]: Great but the rule1 will never match regardless of the priority. You need to make the rule1 's conditions identical to rule2 (replace Lionel with Cristiano on rule1).

Comment (ismailgulek) [reply Tests/SwiftRuleEngineTests/RuleEngineTests/RuleEngineDictionaryTests.swift:382]: I see what you mean, updated.

**Review by santalvarez — APPROVED** [review]
Nice!

**Pull request merged**